File: client/python/gamium/gamium/locator/by.py
from gamium.protocol.generated.Types.ObjectLocatorBy import ObjectLocatorBy


class By:
    def __init__(self, by: int, str: str):
        self.by = by
        self.str = str

    @staticmethod
    def path(str: str):
        return By(ObjectLocatorBy.Path, str)

    @staticmethod
    def tag(str: str):
        return By(ObjectLocatorBy.Tag, str)

File: client/typescript/gamium/src/common/locator/by.ts
import { ObjectLocatorBy } from '../protocols/generated';

export class By {
  constructor(public readonly by: ObjectLocatorBy, public readonly str: string) {}
  static path(str: string): By {
    return new By(ObjectLocatorBy.Path, str);
  }
  static tag(str: string): By {
    return new By(ObjectLocatorBy.Tag, str);
  }
}
